src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -832,5 +819,18 @@ internal static int GetOverloadResolutionPriority(this Symbol symbol)
             Debug.Assert(symbol is MethodSymbol or PropertySymbol);
             return symbol is MethodSymbol method ? method.OverloadResolutionPriority : ((PropertySymbol)symbol).OverloadResolutionPriority;
         }
+
+        internal static bool IsExtensionParameter(this ParameterSymbol parameter)
+        {
+            return parameter.ContainingSymbol is TypeSymbol { IsExtension: true };
+        }
+
+        internal static bool IsExtensionParameterImplementation(this ParameterSymbol parameter)
+        {
+            Debug.Assert(parameter.ContainingSymbol.IsDefinition);
+            return parameter.ContainingSymbol is SourceExtensionImplementationMethodSymbol implementationMethod
+                && !implementationMethod.UnderlyingMethod.IsStatic
```

Comment:
> && !implementationMethod.UnderlyingMethod.IsStatic [](http://example.com/codeflow?start=15&length=51)

It looks like this fix was not covered by a corresponding test. Was it not possible to observe the incorrect behavior without a direct API test?

Reply:
Will add a test